build_projects/shared-build-targets-utils/Utils/SharedFrameworkPublisher.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Newtonsoft.Json;
+
+using static Microsoft.DotNet.Cli.Build.Framework.BuildHelpers;
+using static Microsoft.DotNet.Cli.Build.FS;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class SharedFrameworkPublisher
+    {
+        public static string s_sharedFrameworkName = "Microsoft.NETCore.App";
+        
+        private string _sharedFrameworkTemplateSourceRoot;
+        private string _sharedFrameworkNugetVersion;
+        private string _sharedFrameworkRid;
+        private string _sharedFrameworkSourceRoot;
+        private string _repoRoot;
+        private string _corehostLockedDirectory;
+        private string _corehostLatestDirectory;
+
+        private Crossgen _crossgenUtil = new Crossgen(DependencyVersions.CoreCLRVersion);
+        private string _corehostPackageSource;
+
+        public SharedFrameworkPublisher(
+            string repoRoot, 
+            string corehostLockedDirectory,
+            string corehostLatestDirectory,
+            string corehostPackageSource,
+            string sharedFrameworkNugetVersion)
+        {
+            _repoRoot = repoRoot;
+            _corehostLockedDirectory = corehostLockedDirectory;
+            _corehostLatestDirectory = corehostLatestDirectory;
+            _corehostPackageSource = corehostPackageSource;
+
+            _sharedFrameworkTemplateSourceRoot = Path.Combine(repoRoot, "src", "sharedframework", "framework");
```

Comment:
I am trying to keep this project the same between the two repos to make it easy to propagate changes.